napari/layers/utils/layer_utils.py

```diff
@@ -204,81 +216,485 @@ def _nanmax(array):
 
     if min_val == max_val:
         min_val = 0
         max_val = 1
-    return (float(min_val), float(max_val))
+    return float(min_val), float(max_val)
+
+
+def _get_1d_slices(shape, chunk_size):
+    """
+    Calculate data range in case of 1D data.
+
+    Parameters
+    ----------
+    shape: Sequence[int]
+        Shape of the data either in pixels or chunks.
+    chunk_size: Sequence[int]
+        The size in pixels per dimension of the chunks if data is lazy.
+
+    Returns
+    -------
+    slices: list[slice]
+        Slices to be taken from data for calculating contrast limits.
+    """
+    # If data is very large we only aim to take data around the center.
+    center = shape[0] // 2 * chunk_size[0] if chunk_size else shape[0] // 2
+
+    if not chunk_size:
+        return [
+            slice(
+                center - int(PIXEL_THRESHOLD // 2),
+                center + int(PIXEL_THRESHOLD // 2),
+            )
+        ]
+
+    chunk_size_product = np.prod(chunk_size)
+    allowed_chunks = int(PIXEL_THRESHOLD // chunk_size_product)
+    multiplier = min(allowed_chunks, MAX_NUMBER_OF_CHUNKS)
+
+    return [slice(center, center + chunk_size[0] * multiplier)]
+
+
+def _get_blocks_grid_shape(
+    data_shape: Sequence[int], chunk_size: Sequence[int]
+) -> Tuple[int, ...]:
+    """
+    Get the block shape: the approximate layout of the grid of array chunks.
+
+    Gets the shape of the array not on the pixel level but on the chunk level. For example if x is 100 and
+    chunk size is 10, 10 chunks fit in 100 and will thus be returned. In case of x being 105, still 10 will be
+    returned. This to ensure that only whole chunks will be loaded.
+
+    Parameters
+    ----------
+    data_shape: tuple[int, ...]
+        The shape of an array of chunked data.
+    chunk_size: tuple[int, ...]
+        The size per dimension of the chunks in the chunked data array.
+
+    Returns
+    -------
+    tuple[int, ...]
+        shape indicating the number of chunks per dimension.
+    """
+    return tuple(
+        data_shape[i] // chunk_size[i] for i in range(len(data_shape))
+    )
+
+
+def _get_plane_indices(
+    shape: Sequence[int], offset: int
+) -> List[Tuple[int, ...]]:
+    """
+    Get the indices that correspond to the lowest, middle and highest index of the non-visible dimensions in shape.
+
+    In case the different planes are the same, duplicates are removed.
+
+    Parameters
+    ----------
+    shape: Sequence[int]
+        Either the shape of the raw data (in pixels) or the block shape of the data (chunks)
+    offset: int
+        Number of visible dimensions
+
+    Returns
+    -------
+    idxs: list[tuple[int, ...]]
+        Bottom, middle and top plane (in pixels or chunks) for each non-visible dimension or single plane
+    """
+    bottom_plane_idx = (0,) * (len(shape) - offset)
+    middle_plane_idx = tuple(s // 2 for s in shape[:-offset])
+    top_plane_idx = tuple(s - 1 for s in shape[:-offset])
```

Comment:
```suggestion
    bottom_plane_idx = tuple(int(s * 0.25) for s in shape[:-offset])
    middle_plane_idx = tuple(s // 2 for s in shape[:-offset])
    top_plane_idx = tuple(int(s * 0.75) for s in shape[:-offset])
```
Maybe not using bottom and top plane but some planes from middle of data.